### PR TITLE
Crystal field multispectral fitting function.

### LINKED ISSUE #15701: Crystal field: multispectral fitting function.
Write a multi-dataset fitting function to fit multiple spectra. The crystal field parameters will be global others (such as temperature, etc, line shape parameters) will be local.

This is a part of #15684

### PR DESCRIPTION
**Pull request by mantid-roman**
Added Fitting function CrystalFieldMultiSpectrum.

**To test:**

Run this script. It shouldn't crash.

```
import numpy as np

# Build a reference data set
fun = 'name=CrystalFieldMultiSpectrum,Ion=Ce,Temperatures=(44, 150),ToleranceIntensity=0.001,B20=0.37737,B22=3.9770,B40=-0.031787,B42=-0.11611,B44=-0.12544'
fun += ',f0.f1.FWHM=1.6,f0.f2.FWHM=2.0,f0.f3.FWHM=2.3,f1.f1.FWHM=1.6,f1.f2.FWHM=2.5,f1.f3.FWHM=3,f1.f4.FWHM=1'

# This creates a (empty) workspace to use with EvaluateFunction
x = np.linspace(0, 55, 200)
y = x * 0
e = y + 1
ws = CreateWorkspace(x, y, e)

# The calculated data will be in 'data', WorkspaceIndex=1
EvaluateFunction(fun, InputWorkspace=ws, InputWorkspace_1=ws, OutputWorkspace='data')

# Change parameters slightly and fit to the reference data
fun = 'name=CrystalFieldMultiSpectrum,Ion=Ce,Temperatures=(44, 150),ToleranceIntensity=0.001,Symmetry=C2v,B20=0.37,B22=3.9,B40=-0.03,B42=-0.11,B44=-0.12'
fun += ',f0.f1.FWHM=2,f0.f2.FWHM=2,f0.f3.FWHM=2,f1.f1.FWHM=2,f1.f2.FWHM=2,f1.f3.FWHM=2,f1.f4.FWHM=2'
fun += ',ties=(B60=0,B62=0,B64=0,B66=0,BmolX=0,BmolY=0,BmolZ=0,BextX=0,BextY=0,BextZ=0)'

# (set MaxIterations=0 to see the starting point)
Fit(fun, InputWorkspace='Workspace_0', WorkspaceIndex=1,
           InputWorkspace_1='Workspace_1', WorkspaceIndex_1=1,
           Output='fit',MaxIterations=100, CostFunction='Unweighted least squares')
# Using Unweighted least squares fit because the data has no errors.
```

Fixes #15701.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/CurveFitting/test/Functions/CrystalFieldMultiSpectrumTest.h [hunk 1]
```diff
@@ -99,11 +101,10 @@ class CrystalFieldMultiSpectrumTest : public CxxTest::TestSuite {
...
     alg->execute();
 
-    IFunction_sptr fun = alg->getProperty("Function");
-    for (size_t i = 0; i < fun->nParams(); ++i) {
-      std::cerr << "Param " << i << ' ' << fun->parameterName(i) << " = "
-                << fun->getParameter(i) << std::endl;
-    }
+    auto out0 = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("Workspace_0");
+    TS_ASSERT(out0);
+    TS_ASSERT_EQUALS(out0->getNumberHistograms(), 3);
```
Comment (KarlPalmen) [inline Framework/CurveFitting/test/Functions/CrystalFieldMultiSpectrumTest.h]: Now it looks like a test.

**Comment by KarlPalmen** [comment]
I don't see any change of the release notes.

**Comment by mantid-roman** [comment]
It's in direct inelastic: docs/source/release/v3.8.0/direct_inelastic.rst

**Comment by KarlPalmen** [comment]
Script works without error and produces five workspaces. :shipit:

**Comment by FedeMPouzols** [comment]
There's probably been some issue in the last merge. The pylint job failed because of a compilation error. I looks like there is a `>>>>>>> origin/master` line left in the source files.

**Comment by KarlPalmen** [comment]
Script still works :shipit:

**Pull request merged**